proj-android/PowerAuthLibrary/src/main/java/io/getlime/security/powerauth/networking/client/HttpRequestHelper.java

```diff
@@ -0,0 +1,363 @@
+        } else {
+            // Acquire the encryptor from the helper and keep it locally.
+            // We will use it later for the response decryption.
+            encryptor = helper.getEciesEncryptor(endpoint.getEncryptorId());
+            if (encryptor == null) {
+                // This typically happens when
+                throw new PowerAuthErrorException(PowerAuthErrorCodes.PA2ErrorCodeEncryptionError, "Encryptor creation failed.");
+            }
+            requestData = serialization.encryptObject(requestObject, encryptor);
+
+            // Set encryption HTTP header, only if this doesn't collide with the signature.
+            // We don't sent the encryption header together with the signature header. The reason
+            // for that is fact, that signature header already contains values required for
+            // decryption on the server.
+            if (!needsSignature) {
+                final ECIESMetaData metadata = encryptor.getMetaData();
+                requestHeaders.put(metadata.getHttpHeaderKey(), metadata.getHttpHeaderValue());
+            }
+        }
+
+        // Sign data if requested
+        if (needsSignature) {
+            final boolean available = endpoint.isAvailableInProtocolUpgrade();
+            final PowerAuthAuthorizationHttpHeader header = helper.getAuthorizationHeader(available, requestData, requestMethod, endpoint.getAuthorizationUriId(), authentication);
+            if (header.getPowerAuthErrorCode() != PowerAuthErrorCodes.PA2Succeed) {
+                if (header.getPowerAuthErrorCode() == PowerAuthErrorCodes.PA2ErrorCodePendingProtocolUpgrade) {
+                    throw new PowerAuthErrorException(header.getPowerAuthErrorCode(), "Request is temporarily unavailable, due to pending protocol upgrade.");
+                }
+                throw new PowerAuthErrorException(header.getPowerAuthErrorCode());
+            }
+            // Keep authorization header
+            requestHeaders.put(header.getKey(), header.getValue());
+        }
+
+        // Finalize headers
+        requestHeaders.put("Content-Type", "application/json");
+        requestHeaders.put("Accept", "application/json");
+
+        // Return object with all information prepared for request processing.
+        return new RequestData(requestUrl, requestMethod, requestHeaders, requestData);
+    }
+
+
+    /**
+     * Build a response object from provided data. In case of error throws {@link FailedApiException},
+     * {@link ErrorResponseApiException} or {@link PowerAuthErrorException} exceptions.
+     *
+     * @param responseCode HTTP response status code
+     * @param responseData Response bytes
+     * @return object created from response bytes
+     * @throws Throwable if a deserialization, or decryption error occured.
+     */
+    @Nullable
+    public TResponse buildResponse(int responseCode, @Nullable byte[] responseData) throws Throwable {
+
+        if (responseCode / 100 != 2) {
```

Comment:
This looks a bit weird. We can rely on our backend to return the 200 status code, not sure why we need to consider other 2xx codes?

Reply:
This is copied from old `PA2Client` code.

Reply:
OK, I'll unify this with iOS impl. where only `200` code means "OK"